.github/workflows/build.yml

```diff
@@ -359,15 +359,13 @@ jobs:
 
   release:
     name: Release
-
     if: startsWith(github.ref, 'refs/tags/') || github.event_name == 'workflow_dispatch' && github.event.inputs.publish == 'true'
-
     runs-on: ubuntu-latest
-    environment: Publish
+    environment: pypipublish
```

Comment:
The environment needs to match what is set in truted publishing on PyPI. Unfortunately, it can do *only* lower case environments, so the env in github needs to be all lowercase too.

Reply:
I'm really glad you found this!